src/azure-openapi-linter/rules/ArraySchemaMustHaveItems.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from '../rule';
+export const ArraySchemaMustHaveItems: string = "ArraySchemaMustHaveItems";
+
+const jp = require('jsonpath');
+
+rules.push({
+  id: "S2009",
+  name: ArraySchemaMustHaveItems,
+  severity: "warning",
+  category: "SDKViolation",
+  mergeState: MergeStates.individual,
+  openapiType: OpenApiTypes.arm,
+  appliesTo_JsonQuery: "$.definitions.*.properties.*.type",
+  run: function* (doc, node, path) {
+    const msg: string = "Please provide an items property for array type: ";
+    if (node === 'array') {
+      // climb up the json path to get hold of the parent element
+      // check if the parent element has an items node defined
+      const propObject = jp.query(doc, path.splice(0, path.length - 2).join('.'));
```

Comment:
why path.length-2? it'd be good to get a comment before this line with an example of what we're looking to accomplish, so we can interpret the code more easily.